lib/models/asyncapi.js

```diff
@@ -196,6 +208,69 @@ class AsyncAPIDocument extends Base {
   hasCircular() {
     return !!this._json[String(xParserCircle)];
   }
+
+  /**
+   * @callback TraverseSchemas
+   * @param {Schema} schema
+   * @param {SchemaIteratorCallbackType} callbackType
+   * @param {String} propName if the schema is from a property get the name of such
+   * @returns {boolean} should it continue deeper
+   * 
+   */
+
+  /**
+   * Traverse schemas in the document and select which types of schemas to include.
+   * By default all schemas are iterated
+   * 
+   * @param {TraverseSchemas} cb 
+   * @param {SchemaTypesToIterate[]} schemaTypesToIterate
+   */
+  traverseSchemas(cb, schemaTypesToIterate) {
+    traverseAsyncApiDocument(this, cb, schemaTypesToIterate);
+  }
+}
+
+/**
+ * Marks all recursive schemas as recursive.
+ * 
+ * @private
+ * @param {AsyncAPIDocument} doc 
+ */
+function markCircularSchemas(doc) {
+  const seenObj = [];
+  const lastSchema =  [];
+
+  //Mark the schema as recursive
+  const markCircular = (schema, prop) => {
+    if (schema.type() === 'array') return schema.json()[String(xParserCircle)] = true;
+    const circPropsList = schema.json()[String(xParserCircleProps)] || [];
+    if (prop !== undefined) {
+      circPropsList.push(prop);
+    }
+    schema.json()[String(xParserCircleProps)] = circPropsList;
+  };
```

Comment:
I can see what you mean, however this reference are only created each time you create a new instance of `AsyncAPIDocument` and are never used since. So moving it outside does not create any optimization in this case :)

Reply:
You are right, sorry :)